pybamm/models/base_model.py

```diff
@@ -195,33 +197,46 @@ def update(self, *submodels):
     def check_well_posedness(self):
         """
         Check that the model is well-posed by executing the following tests:
-        - All the variables that appear in the rhs and algebraic equations appear in
-        the rhs keys, with leeway for exactly n variables where n is the number of
-        algebraic equations. Overdetermined if more equations than variables,
-        underdetermined if more variables than equations.
+        - Model is not over- or underdetermined, by comparing keys and equations in rhs
+        and algebraic. Overdetermined if more equations than variables, underdetermined
+        if more variables than equations.
         - There is an initial condition in self.initial_conditions for each
         variable/equation pair in self.rhs
         - There are appropriate boundary conditions in self.boundary_conditions for each
```

Comment:
We should also check that there are appropriate boundary conditions for the variable/equation pair in self.rhs